core/trino-main/src/main/java/io/trino/server/PluginManager.java

```diff
@@ -191,10 +192,12 @@ private void installPluginInternal(Plugin plugin, Supplier<ClassLoader> duplicat
             accessControlManager.addSystemAccessControlFactory(accessControlFactory);
         }
 
-        for (PasswordAuthenticatorFactory authenticatorFactory : plugin.getPasswordAuthenticatorFactories()) {
-            log.info("Registering password authenticator %s", authenticatorFactory.getName());
-            passwordAuthenticatorManager.addPasswordAuthenticatorFactory(authenticatorFactory);
-        }
+        passwordAuthenticatorManager.ifPresent(authenticationManager -> {
```

Comment:
We shouldn't need to make this change. There's no config being read here -- it's simply registering the names of the password authenticators. This change would make password authenticators different from all the other plugin services.